uniffi_bindgen/src/bindings/kotlin/templates/Helpers.kt

```diff
@@ -28,3 +28,80 @@ inline fun <T : FFIObject, R> T.use(block: (T) -> R) =
             // swallow
         }
     }
+
+internal typealias Handle = Long
+internal class ConcurrentHandleMap<T>(
+    private val leftMap: MutableMap<Handle, T> = mutableMapOf(),
+    private val rightMap: MutableMap<T, Handle> = mutableMapOf()
+) {
+    private val lock = java.util.concurrent.locks.ReentrantLock()
```

Comment:
IIUC, this lock only protects concurrent access to the maps, and we release it before invoking the callback. So, it's possible to have multiple callbacks being dispatched at the same time. Accurate?

Reply:
Yes, correct.